mypy/nodes.py

```diff
@@ -1649,6 +1859,46 @@ def __str__(self) -> str:
                             ('Names', sorted(self.names.keys()))],
                            'TypeInfo')
 
+    def serialize(self) -> Union[str, JsonDict]:
+        data = {'.class': 'TypeInfo',
+                'fullname': self.fullname(),
+                'alt_fullname': self.alt_fullname,
+                'subtypes': [t.serialize() for t in self.subtypes],
+                'names': self.names.serialize(self.alt_fullname or self.fullname()),
+                'defn': self.defn.serialize(),
```

Comment:
Maybe add comment mentioning that this is where all `ClassDef` objects come from so there shouldn't be duplicates.